.github/workflows/main.yml

```diff
@@ -51,12 +51,32 @@ jobs:
       - run: pyright
         if: success() || failure()
 
+  ##
+
   test:
     runs-on: ubuntu-latest
-    needs: setup-default-env
+    strategy: 
+      fail-fast: false
+      matrix:
+        python: [py38, py39, py310, py311] 
+        pytorch: [torch20, torch21, torch22, torch23, torch24]
+
     steps:
       - uses: actions/checkout@v4
+
+      - name: Inject dependencies into pixi.toml
+        run: |
+          sed -i '/\[feature.dev.pypi-dependencies\]/i \
+          [feature.test.dependencies]\
+          python = "3.8"\
+          [feature.test.pypi-dependencies]\
+          torch = "2.3"\n' pixi.toml
```

Comment:
Please see my changes and comments! These version numbers are currently/temporarily hardcoded (and I think also incorrectly formatted).